### ISSUE TITLE
need help! difficulties training simple network

### ISSUE THREAD

**Issue by merckxiaan** [FluxML/Flux.jl#331]
Hello I'm trying to create a model to classify rgb values as a color. I'm new to this stuff and I'm unable to make my simple model work.

This is the form of my data:

the RGB color values:
![xs](https://user-images.githubusercontent.com/31353884/43002665-dcd4555e-8c29-11e8-8dcf-e4f441cd2c32.png)

onehot-vectors for the colors:
![ys](https://user-images.githubusercontent.com/31353884/43002752-2fdbd8f8-8c2a-11e8-8cba-0498fd4f97d7.png)

I've zipped the xs and ys together to feed it to my model. I arbitrarily chose some layer sizes to test things out:
```
model = Chain(Dense(3, 20, relu), Dense(20, 20, relu), Dense(20, 20, relu), Dense(20, 9), softmax)
loss = (x, y) -> crossentropy(model(x), y)
opt = () -> SGD(params(model))
callback = () -> @show(loss(xs[37], ys[37]))
```

When I run the model for 10 epochs, the loss on one random datapoint doesn't change:
`@epochs 10 Flux.train!(loss, data, opt, cb = throttle(callback, 5));`
![training](https://user-images.githubusercontent.com/31353884/43003025-3b40b35c-8c2b-11e8-8866-af8780fd05bd.png)
I'm sure I'm making a stupid mistake, but I can't seem to figure out where.

Any help would be appreciated,

Jules

my ipynb and data:
[color_classifier.zip](https://github.com/FluxML/Flux.jl/files/2213922/color_classifier.zip)
